lib/src/transactions.dart

```diff
@@ -82,7 +82,7 @@ class ArweaveTransactionsApi {
 
     while (!uploader.isComplete) {
       if (!dryRun) {
-        await uploader.uploadChunk();
+        await uploader.batchUploadChunks();
       }
       yield uploader;
```

Comment:
Is the crummy UI performance happening because we need the full batch of 100 chunk uploads to complete before the uploader yields?

Reply:
Yes. That is the case

Reply:
Alternatively we could let upload chunks run without awaiting it, but have a different periodic yield of uploader that will stop if it sees that the upload is complete.

Reply:
I believe the answer is yes. Could we make batchUploadChunks be a stream of progress updates that yields each time a chunk completes uploading? We'd run the stream to exhaustion before calling batchUploadChunks again? I'm not super well versed in dart streams yet, but I think if we did that, we could do an `await for` on the stream, and each time it emits, we could yield the uploader back to the UI for presentation of progress. If we think the UI will have trouble debouncing that many cubit event emissions, you could do it every time the upload percentage has advanced by 1% or something. Or every 20 chunks. Something like that.

Reply:
Another thought: the shortcoming of this approach is that each batch is really only as fast as its slowest chunk upload. It would be nice to just continuously start new chunk uploads as each one finishes.

Reply:
Created a PR with these changes here https://github.com/ardriveapp/arweave-dart/pull/16